torchdata/datapipes/iter/transform/bucketbatcher.py

```diff
@@ -42,44 +42,32 @@ class BucketBatcherIterDataPipe(IterDataPipe[DataChunk[T_co]]):
     in_batch_shuffle: bool
     length: Optional[int]
 
-    def __init__(
-        self,
+    def __new__(
+        cls,
         datapipe: IterDataPipe[T_co],
         batch_size: int,
         drop_last: bool = False,
         batch_num: int = 100,
         bucket_num: int = 1,
         sort_key: Optional[Callable] = None,
         in_batch_shuffle: bool = True,
-    ) -> None:
+    ):
         assert batch_size > 0, "Batch size is required to be larger than 0!"
         assert batch_num > 0, "Number of batches is required to be larger than 0!"
         assert bucket_num > 0, "Number of buckets is required to be larger than 0!"
-        super().__init__()
 
-        # TODO(136): Verify _datapippe is not going to be serialized twice and is able to reconstruct
-        self._datapipe: IterDataPipe[T_co] = datapipe
```

Comment:
`self._datapipe` is only used to get `__len__`. And, having it would break the expected DataLoader graph. As `IterDataPipe` doesn't guarantee to have length, removing it and `__len__` function from `bucketbacher`.

Reply:
We may be able to rely on graph to generate proper length in the future.

Reply:
Yeah. But, I believe this is not a mandatory feature as we should not let users expecting that `len` is always available using `IterDataPipe`.